### ISSUE TITLE
Waiting for this message, this may take a while.

### ISSUE THREAD

**Issue by 532910** [element-hq/element-android#1721]
Phone was updated from the latest stable riot-android to element via GPlay (no beta subscription)

Now all messages in all e2ee direct chats including new messages are: `Waiting for this message, this may take a while.`

After update the phone was verified via pitcture sequence with riot-web.
All contacts are verified and has green point/shield on all devices, all contacts are on my own synapse instance (1.15.1-1 debian stable).

![image](https://user-images.githubusercontent.com/14617699/87842291-0dd5b800-c8b4-11ea-990d-d4b4174fff30.png)

Cross-signing is enables, Keys are trusted, Private keys are not known:
![image](https://user-images.githubusercontent.com/14617699/87842340-876da600-c8b4-11ea-9fb9-5b87351e3835.png)

I tried to clear cache and to restart phone without sucess.

**Comment by 532910** [comment]
Trying to re-verify (Secure Backup -> Set up on this device) this phone gives:

via QR code both devices shows error:
![image](https://user-images.githubusercontent.com/14617699/87842723-3ca15d80-c8b7-11ea-9717-9bce46b153a5.png)

via image set the phone shows:
![image](https://user-images.githubusercontent.com/14617699/87842743-75d9cd80-c8b7-11ea-91d2-dfa3c5f0a28e.png)
riot-web says:
```
Verify other session
Verification cancelled

You cancelled verification. Start verification again from the notification.
[Clear]
```

**Comment by 532910** [comment]
Restoring messages from backup decrypts all messages. Newly received messages after backup recovery are shown as `Waiting for this message, this may take a while.` Restoring messages from backup again decrypts all messages, even just received before this restore.

**Comment by 532910** [comment]
Is any update there? It's completely impossible to use Element after upgrade!

**Comment by 532910** [comment]
Right now, after restoring messages from backup it says:
```
Cross-Signing is enabled
Private Keys on device
```
But still shows new messages as `Waiting for this message, this may take a while`

**Comment by 532910** [comment]
Nothing has changed after more than a week of waiting, so I had to reinstall Element.

**Comment by PureTryOut** [comment]
My parents have the same issue when talking between them. They do resolve after 24 hours or more, but of course that makes it impossible to have a proper conversation. Please give this issue some priority, beside the vague text it's more than just an UX bug!

**Comment by BillCarsonFr** [comment]
Hello, can you please give more information on the element version of both devices?

Also could you check the security settings on both devices (is Encrypt to verified session only is ON?)
<img width="381" alt="image" src="https://user-images.githubusercontent.com/9841565/89193231-fda52480-d5a5-11ea-9ada-472178257b7d.png">

Do you have other sessions?
The fact that key backup import is able to decrypt could mean that your other session are getting the keys and uploading them to backup.
So it could be possible that your device is not getting the keys.

Go to settings > Advanced > and enable developper mode, this will allow you to see more about the error, if you go back to the room, what do you see for the errors?

Did you submit some rageshake?

**Comment by PureTryOut** [comment]
Speaking for my parents here, as I don't have this issue myself.

They use the latest version from the Google Play store, so whatever version that is. They don't have "Encrypt to verified session only" on.
They have other sessions, for both a tablet and a laptop (so webbrowser) at least.

I'll ask them to enable developer mode and submit a rageshake. Thanks for looking in to it!

**Comment by 532910** [comment]
> element version

the latest available on GPlay

Encrypt to verified session only is OFF.

> have other sessions?

Yes.

**Comment by MrCustomizer** [comment]
A few users on my server have reported this issue to me. Seems like this is still happening. Any news on workarounds/solutions?

**Comment by aczerniak** [comment]
It happened to me today after I cleared cache on Element installed on Android phone. I use Element 1.0.9[40100092]. In developer mode message says that sender's device didn't send keys. Before I cleared cache those messages were properly decrypted. I have 4 sessions (3x Firefox and one phone - the issue affected phone only), 2 of them are open and devices that were used sent affected messages are also on so at very least they should resent needed keys.

I don't have cross signing enabled. I verify all session manually.

**Comment by thomashoullier** [comment]
Here on `matrix-synapse-py3 1.22.1` (server), a user on the android client (update 19 Oct 2020) reported the same problem.
"Waiting for this message, this might take a while" for all the messages from a particular user starting at some date in e2e encrypted channels. The user with the bug can still receive messages from other users on the same channels.

**Comment by lasselin** [comment]
A user just reported me this issue on the matrix.org server.

**Comment by akallabeth** [comment]
Have the same issue from time to time with different users (and some small group chats).
This did not happen with 1.0.8 and earlier (all from F-Droid)
All accounts on matrix.org.

**Comment by austinbutler** [comment]
Getting this for the first time today. Didn't change anything and was working fine yesterday. Encrypt to verified sessions only is off.

![Screenshot_20201120-204111](https://user-images.githubusercontent.com/354741/99867846-2ca1cf00-2b72-11eb-8c73-09c09ffd4c20.png)

**Comment by gtirloni** [comment]
I'm getting this one a newly activated device after I did the cross-sign verification (Element Android 1.0.11 [40100112] (G-b847)).

**Comment by smeyersdev** [comment]
My wife also runs into this bug on her Element installation on Android (Play Store). There is a strange behavior: As long as she stays at home connected to Wifi network, the issue does not appear. But as soon as the smartphone is using mobile LTE internet, it happened several times that I received these messages from her which never got decrypted.

**Comment by jrozanski** [comment]
Indeed @smeyersdev, I observed the same behaviour sporadically on poor quality connections. It appears to be related to instability of the connection.

**Comment by dalcde** [comment]
This seems to depend on the session sending the message. In one of my conversations, I have interleaving messages from different devices (same user), and only the messages from one of the devices are stuck in "Waiting for this message, this may take a while".

**Comment by dalcde** [comment]
I'd imagine the issue starts with an attempt to retrieve a key failing. What is the retry strategy in this case? In other parts of the matrix ecosystem I've had issues with exponential back-offs leading to long delays when something is inaccessible.

**Comment by michel-slm** [comment]
I'm seeing this too, and the weird thing is:
- it happens with messages from one user to a group with multiple users
- messages from other users are fine
- messages from this user only get delayed on one of my Android phones (Pixel 3, Android 12 beta 2) but not on the other (Pixel 5, Android 11)
- this only started today, previously both phones are fine
- 1:1 messages with this user is fine on both phones
We tried verifying our sessions, it does not seem to have any effect. And once they send further messages, the previously "waiting" messages seem to clear up, but that means I'm always unable to see their last message on this thread.

**Comment by Wodann** [comment]
I'm consistently seeing this issue on my android phone for the last message sent by one user. The moment they send more messages, the message that previously displayed the error will become visible, but the new latest message will report the error instead.

**Comment by jzacsh** [comment]
regarding the theory about mobile connectivity being the root cause: while I **am** seeing a correlation with mobile (non-wifi) usage, I think the bigger problem is the undefined behavior from then on.

That is, the last couple times I hit this bug, I then had trouble working _around_ it. For example we'd see these steps:

1. hit the bug in 1:1 encrypted chat, while out on mobile connection
2. both parties jump onto _same_ (fast) wifi LAN
   1. both keep trying to send chats
   2. both showing phones in person (one person has all messages rendered, the other sees the bug on messages -2, -3, -4 (but 0, -1 sent over wifi are fine))
3. give up on element for a day or so
4. over a day later one person reports (via Signal) that _some_ of the messages are suddenly showing

step (2) involves _both_ parties having their Element app out, actively looking at the same 1:1 room in the foreground on Android, actively connected on a fast wifi connection. So:
- a) if it _is_ failed request (needing retry), perhaps this is popping up due to a retry bug?
- b) Or perhaps this is a ux issue? (eg just need to add a _"retry **now** [i promise i'm on good connection now]"_?)

**Comment by agherzan** [comment]
I have the same issue: iPhone client vs Android client. Only some of the messages from iPhone end up in the "waiting"  state on the Android side. What I saw, is that sending more messages starts to refresh previous messages in this state. In other words, the last message is always in "waiting" and receiving more messages unlocks it while locking the next one. It also happens only with this iPhone user. I have chat rooms, with clients on exactly same platform and room configuration where this doesn't reproduce ever.

**Comment by Alkaid-Benetnash** [comment]
I have the same issue: android element A <- latest docker synapse -> android element B.
A cannot see messages from B timely about 2weeks ago. Destroy and recreate 1-to-1 room solved the issue.
Now B cannot see messages from A timely. Want to look for a workaround without destroying the existing room.

Any updates? Is this still bottlenecked by collecting bug reports?

**Comment by Wodann** [comment]
> regarding the theory about mobile connectivity being the root cause

In my case I'm actually not using mobile data at all. Whenever I receive messages, it's on wifi.

**Comment by begincalendar** [comment]
I just signed out of the app, cleared the app's storage (all of it, not just the cache) for good measure and then signed back in.
Seems like it might have worked around the issue.

**Comment by newptcai** [comment]
I logged out to resolve this issue and now almost all my messages are turned into "Waiting for this messages..."

**Comment by crashbandicode** [comment]
> Phone was updated from the latest stable riot-android to element via GPlay (no beta subscription)
> 
> Now all messages in all e2ee direct chats including new messages are: `Waiting for this message, this may take a while.`
> 
> After update the phone was verified via pitcture sequence with riot-web.
> All contacts are verified and has green point/shield on all devices, all contacts are on my own synapse instance (1.15.1-1 debian stable).
> 
> ![image](https://user-images.githubusercontent.com/14617699/87842291-0dd5b800-c8b4-11ea-990d-d4b4174fff30.png)
> 
> Cross-signing is enables, Keys are trusted, Private keys are not known:
> ![image](https://user-images.githubusercontent.com/14617699/87842340-876da600-c8b4-11ea-9fb9-5b87351e3835.png)
> 
> I tried to clear cache and to restart phone without sucess.

So I am also experiencing this issue on Android with a second session on iOS. I notice you have “setup secure backup” in your photo instead of “reset secure backup”. As a potential workaround on Android, try tapping the “setup secure backup” option and walking through the verify process. This seems to have solved the issue for me. Depending on the situation I may have to open another session to get the “this may take a while” to go away but I only have to do that once per user. I think the keys get messed up somehow. Hopefully this workaround works for someone.

**Comment by bahur142** [comment]
> I'm consistently seeing this issue on my android phone (works fine in my desktop browser's session) for the last message sent by one user. The moment they send more messages, the message that previously displayed the error will become visible, but the new latest message will report the error instead.
> 
> I've tried clearing the cache, but to no avail.

Exactly the same situation between me and a friend using self hosted services!

**Comment by dessalines** [comment]
Just ran into this problem today. Newest version on android, matrix.org homeserver. 

Anything I can do to fix?

**Comment by crashbandicode** [comment]
Another thing to try is installing v1.1.9 from F-droid. The one from Aurora Store had decryption issues but downgrading to that version specifically from F-droid seems to have fixed some of these cases.

**Comment by kernitus** [comment]
Same issue, I found if I enable developer mode the error message becomes `Unable to decrypt: OLM`

**Comment by kat-co** [comment]
Try issuing a `/discardsession` command on the client _not_ experiencing the issue. That seems to have cleared things up for me.

I think the megolm session between devices becomes messed up and element doesn't detect the issue. This should create a new session between devices.

**Comment by benmordecai** [comment]
Someone I talk to is reporting this issue as well.

**Comment by jtrees** [comment]
> Try issuing a `/discardsession` command on the client _not_ experiencing the issue.

Unfortunately this isn't an option on iOS.

**Comment by kernitus** [comment]
> Try issuing a `/discardsession` command on the client _not_ experiencing the issue. That seems to have cleared things up for me.
> 
> I think the megolm session between devices becomes messed up and element doesn't detect the issue. This should create a new session between devices.

Actually this did solve the issue. Now I wonder why it's happening in first place :thinking:

**Comment by finzzz** [comment]
> Another thing to try is installing v1.1.9 from F-droid. The one from Aurora Store had decryption issues but downgrading to that version specifically from F-droid seems to have fixed some of these cases.

This doesn't fix the issue.

**Comment by uncivilgentleman** [comment]
Happy bug anniversary everyone!

Seriously though this bug is crippling.  An unreliable chat app is a hard sell to friends and family.

**Comment by sexybiggetje** [comment]
I have this same problem with a new device; Followed the steps above (including the discard session). To no avail. I'm running v1.1.12 from fdroid

**Comment by mcg-matrix** [comment]
> An unreliable chat app is a hard sell to friends and family.

I guess a lot of us "unpaid admins and consultants" will agree to that statement.

However, this issue has actually helped me stress the importance of running at least two sessions with enabled Secure Backup. Following that rule has proven beneficial in various situations in which one session was unable to decode something (because of a software issue, accidentally signed out, phone gone lost, ...) but the second session could decode and share the secret.

**Comment by UkeHa** [comment]
For me it worked on my desktop but not my phone (i'm using the latest beta branch on my phone). So i reset the cross-signing keys on my mobile device and reconfirmed my device via the qr code on my desktop. That was the only thing to instantly bring back all messages. Logging out and back in did nothing for me. This worked on a private server with federation on the lastet stable build.

**Comment by dessalines** [comment]
I just encountered this on a 2nd device. This is a critical bug that won't let you use any new login from an android device.

**Comment by benmordecai** [comment]
This issue may singlehandedly encourage my team to not use a Matrix server and to switch to a proprietary solution.

**Comment by crashbandicode** [comment]
Is there a way to get it more prioritized/looked at?

Jul 19, 2021 3:51:35 PM Ben Mordecai ***@***.***>:

**Comment by benmordecai** [comment]
It is already a priority 1 issue. I am not sure what more can be done. I am not sure anyone has even fully identified the cause of the bug, but it is troubling that this bug has been active for over a year with a P1 for about 11 months of that time without any progress.

**Comment by JanZerebecki** [comment]
There are a few things that are a more specific cause than this issue that includes multiple different problems or that can be done to improve this issue:

https://github.com/vector-im/element-android/issues/1721#issuecomment-880214682 reads:
> Try issuing a `/discardsession` command on the client _not_ experiencing the issue. That seems to have cleared things up for me.

That this helps points to one issue.

Another issue is if you ignore someone or enable "Never send encrypted messages to unverified sessions from this session" then they will see this. The UI should make this more obvious which is which on both sides.

https://github.com/vector-im/element-web/issues/2996 has a list of more issues related to this.
I haven't yet read enough of the spec to say if the top level of the protocols related to keys, signatures and encrypted messages might benefit from a demonstration that they can't end up in an unwanted error state.
There might be missing tests both on the server and client https://github.com/matrix-org/matrix-doc/issues/3287 side related to this.
If you want to identify the exact cause of one specific instance of this problem: First note the detailed error code you can see when dev mode is enabled. If that and the first 2 paragraphs of this comment don't already help, maybe writing down all the details of the protocol happenings related to the failed message can help you identify the cause and a workaround.

One way to entirely avoid this issue is to avoid encrypted rooms.

**Comment by dessalines** [comment]
I just lost a family member using element because of this. We'll send a message and it'll get through fine, on multiple devices, then 12 hours later  a new message will be stuck on "waiting for this message...."

**Comment by theduke** [comment]
I also recently stumbled over this a week ago after switching to a new phone.

This is a serious dealbreaker and I regularly see it mentioned in various discussion channels, like Hackernews, so it is definitely a regular occurrence.

I'm quite disappointed that this isn't receiving more attention.

**Comment by anoadragon453** [comment]
Hey all - thanks for sending in your reports. As a result, we noticed a [serious regression](https://github.com/matrix-org/synapse/issues/10456) in Synapse v1.38.0 which caused Element Android to have a significantly higher chance of running into decryption issues.

The issue is now [fixed](https://github.com/matrix-org/synapse/pull/10457) with Synapse v1.38.1, which was released on Thursday. The upcoming Synapse v1.39.0 will fix [another long-standing encryption bug](https://github.com/matrix-org/synapse/pull/10485) which could sometimes cause decryption issues on various clients, including Element Android. As these new versions are rolled out to the federation, encryption issues should become less and less frequent.

We're still tracking down the edge cases though -- as a reminder, the best way to help debug encryption issues when they pop up is to hit the "Submit Debug Logs" from settings (or shake your phone) on any devices that were affected. This allows us to get a clearer picture of why keys didn't arrive or failed to send. Thanks!

**Comment by jzacsh** [comment]
@anoadragon453  thanks for the update. Your reminder, while it might seem obvious, is actually quite helpful to hear. Specifically, I personally hadn't submitted logs in a while, because (in my imagination) no one with logs-access was actively looking into this bug and thus no one analyzing said submissions. I'll start sending again 👍

_(to clarify, in my experience, often such log submissions have a TTL of retention before they're autodeleted, so I was picturing all the comments here by end-users would be accompanied with logs that would be long-since autodeleted by the time someone came along to try to investigate)_

**Comment by bahur142** [comment]
It happens again. With Element 1.1.4 on Synapse 1.39.0

![image](https://user-images.githubusercontent.com/30805725/127855955-5e797e18-00ea-4dbf-a8ba-d89f34403e9c.png)

**Comment by austinbutler** [comment]
@anoadragon453 when should we expect this fix to be rolled out to Element servers? I'm having this issue today.

**Comment by austinbutler** [comment]
I'm well over 24 hours and still have the message on a particular incoming message on my phone. Element Desktop sees the message just fine.

**Comment by anoadragon453** [comment]
@austinbutler The fixes mentioned above have been rolled out to matrix.org and all EMS servers for at least a week. If you are seeing further issues with encryption, I recommend submitting debug logs from both affected clients and tagging this issue in your report so we can have a look -- it's likely to be something other than what was addressed with the above PRs.

**Comment by austinbutler** [comment]
@anoadragon453 sent.

**Comment by dead10ck** [comment]
I still see this message. I sent a bug report

**Comment by matrixes** [comment]
I still see this message on the latest Element and our own selfhosted Matrix server that we updated today (to 1.40.0) and latest clients.

The only discernible patterns I can see is that I'm using Element Android and my buddy is using the Element iOS client, and that's the only chat that has this issue. All others, be it Android<->Web or Web<->Web or Anything<->Anything except Android<->iOS, work without a hitch.

I've sent a bug report. It's about time we figure out a way to squash it once and for all.

I'm willing to do tests and whatnot to move this issue forward to its final resolve.

**Comment by PxEclipse** [comment]
Had this issue often in the past in one on one conversations and private rooms. Today it happened again. The only fix is /upgraderoom. Android to android communication on matrix.org.
I appreciate the effort and that element exists. But for me and the people I use it with, it's not a usable state.

**Comment by dontezuma** [comment]
We have a server and have identified the same problem.  My desktop client displays all messages correctly, but when my partner sends a message through his iPhone, it displays the message. When he sends via laptop, it also displays immediately on the phone. 

So, as is clear from prior posts, this is strictly an issue when someone sends messages FROM and iPhone TO an Android. 

I say this to recommend maybe retitling the bug to be more specific.
![image](https://user-images.githubusercontent.com/46940662/131523697-19df8475-9c50-46ae-b6b4-4ee70bf20791.png)

**Comment by russelldavies** [comment]
I just started seeing this in a room of three participants, all Android devices. I think it happens when there is poor network connectivity and the device sending the message also has to rekey and the new keys are delayed in sending.

**Comment by dontezuma** [comment]
I'd agree with that assessment, but _every_ message from my partner's iPhone have had this delay, even when he's not in a degraded connectivity situation.  Of course, these are not mutually exclusive.  That could very well be happening, and it's a separate cause of the same issue.

**Comment by 9ler** [comment]
Thanks a million for making this software <3
I am seeing this problem as well. On androide phones (androide 10) also when they are on ok WIFI and on the same network as the synaps server. But for some reason its only from one device to the other, not both ways.

**Comment by florian6973** [comment]
Hi!
I have a e2ee room with 4 family members. My mum is seeing this message and it is very problematic. I helped her update Element on Android, clear the cache, but nothing has changed. Is there a fix to this error ?
Thanks.

**Comment by SoftwareSchlosser** [comment]
Same problem here.
Sending from Android to Android + desktop app.
The desktop app always shows the messages but the receiving android app doesn't.

**Comment by qtip** [comment]
I have been seeing this problem for several days now on my android phone. The web app works fine but the android app shows the message.

**Comment by tomasradej** [comment]
Update: I was seeing this message in one specific channel (a p2p with one user). We have closed that channel and opened a new one, and I haven't had this problem since.

**Comment by dawnmew** [comment]
This is CRITICAL. How is this not fixed yet? It's stopping people from seeing each others' DMs... I rely on my phone when I'm out, and this is now the single showstopping bug preventing me from thinking of Element as reliable. I've only started seeing this in the last month or two, but my friends are seeing it too. This is horrific. Do something...?

**Comment by zopieux** [comment]
@dawnmew Please review some basic [etiquette](https://gist.github.com/g1eny0ung/9e7d4d0f72547a8d156452e76fa8f7a3) [materials](https://medium.com/@fommil/the-open-source-entitlement-complex-bcb718e2326d) for interacting with open source initiatives. Thanks.

**Comment by eTomte** [comment]
This issue just happened to me in a 1:1 conversation today.  Messages I sent from Desktop client would give the "Waiting for this message" error on the recipients Android phone on Element.  He could see the same messages when using Element Desktop.

**Comment by frederictobiasc** [comment]
I'm also affected. Is there anything I can provide you with to help debugging this issue?

**Comment by tomasradej** [comment]
I have encountered the problem again in a room with multiple people and a different 1:1 chat.

**Comment by enzingerm** [comment]
I had the same problem with two Android clients, A and B, on version 1.3.1 and Synapse 1.43.0. Always the latest few messages from A to B couldn't be decrypted and it showed "Waiting for this message..." (this was not the case for messages from B to A). Over the course of the next hours the messages would be resolved.

I enabled "Encrypt to verified sessions only" in the settings of A, sent a message, and disabled the option again. Since that, messages can be decrypted immediately on B. Maybe this helps others, or gives the devs a hint what might be the root cause...

**Comment by danger89** [comment]
An Android user (using my synapse server), experience the same problem on her client-side.

![image](https://user-images.githubusercontent.com/628926/137409324-92bd61dc-d38c-44ef-917f-25a18d48132c.png)

**Comment by SoftwareSchlosser** [comment]
For me this also didn't happen anymore for some time. Seems like there has been made at least some improvement.

**Comment by RussellAult** [comment]
I just ran into this issue as well (room with only two participants, messages arriving fine in a web browser, messages arriving with the "waiting for message, this may take a while" error showing up in an up-to-date GPlay build). Here's what I've noticed:

- A "Force stop" and re-opening of Element made all previously-received messages decrypt *except for* the most-recently-received one
- Messages received after the "Force stop" will still show an error, but:
  -  Intermittently, the immediately-next-received message after the "Force stop" will decrypt (even though subsequent ones do not)
  - On at least one occasion, the immediately-next-received message showed an error but its arrival caused the *previous* message (i.e. the last one received before the "Force stop") to decrypt (which seems very strange indeed!)

**Comment by vetko** [comment]
I can also confirm that the solution of @enzingerm is working in my case ("Never send encrypted messages to unverified sessions from this device" ON -> send message -> OFF).  
I have another pair of clients in the same situation, I will try `/discardsession` as the other way, let's see if that solves it as well.

**Comment by kat-co** [comment]
> https://github.com/vector-im/element-android/issues/1721#issuecomment-880214682

I just experienced this for a second time after months of stability. Running `/discardsession` worked once again.

**Comment by danger89** [comment]
These are still workarounds.. The root cause is still unknown and not resolved.

**Comment by aethralis** [comment]
I have the same problem. Closing and opening the android app immediately displays messages that previously were not displayed. However, if other user sends me a new message, then I get "Waiting for this message..." again.

**Comment by lindhe** [comment]
I got this issue now, after the recent update from 1.3.6 to 1.3.7.

**Comment by lindhe** [comment]
Clearing app data and reverifying fixed it for me. This should really be investigated more so people don't have to get into such a state. I cannot remember ever having to do something similar on Telegram (which admittedly is not e2ee in my case).

**Comment by kat-co** [comment]
This is a really annoying thing that happens, but I wanted to point out that this bug report is one of those that probably represents a symptom and not a specific bug. This symptom is probably rooted in disparate underlying bugs.

So different things are probably going to fix this for different people, and it's going to collect a lot of "me too, but for me it happened when...". Unfortunately that means it's going to look like the devs are ignoring a critical issue even if they're working on fixing some of the bugs that underpin this. I.e.: you can't fix and close a symptom until you fix all the bugs that cause it.

It's good that everyone has a place to look for ideas, because maybe they're experiencing the same underlying bug. A different strategy might be to report symptoms as: "When I did this, then I started getting this error". That would help segregate the same symptom into buckets of potential root causes.

Just some thoughts. I hate seeing people come to this issue and get frustrated because it looks like it's being ignored. Good luck to you all.

**Comment by dead10ck** [comment]
At the same time, it only takes a few seconds to report a status update like "I've been searching for the root cause, but I've been having trouble reproducing the issue, so I haven't found it yet."

The best way to avoid the perception of being ignored is to communicate anything at all, even if no real progress has been made.

**Comment by tomasradej** [comment]
@kat-co That's the problem, I wasn't doing anything at all, it just started happening one day. The only thing I can say to narrow it down is that the room(s) where this happens for me have been unencrypted for a long time and then I enabled encryption, but it still took many months for this problem to manifest.

**Comment by zopieux** [comment]
As a counter point to the above, in my setup the bug happens in rooms that were always encrypted from there get go, both private messages and group rooms (not that I believe it's any different in the matrix protocol).

**Comment by lindhe** [comment]
I'm annoyingly good at finding bugs. I'll give it a shot, when I find some time, to reproduce it reliably. It feels to me like it will be possible...

**Comment by tomasradej** [comment]
@lindhe thank you so much. Can I buy you a beer? (Seriously)

**Comment by lindhe** [comment]
Only after I get around to it and if I am successful. :D

**Comment by smeyersdev** [comment]
Happened again. See my post above. Situation was similar to before: 

- Received encrypted messages from my wife, but she could read my messages
- Wife's phone is on mobile LTE internet while mine is on the home Wifi
- Her phone is a Pixel 3 with latest update state and Element installed from Play Store
- My phone is OnePlus 5T running ungoogled LineageOS. Element installed from F-Droid.

2 days before, I used Element Web on PC and deleted all sessions (older web sessions and all Android sessions). Then I logged into Element again on my phone and tablet.

Solved it the following way:
- Go to Androids app settings for Element 
- Force end element
- Clear cache of Element
- Start element again

Then after few seconds, the messages were decrypted.

**Comment by chr-1x** [comment]
Happened to my partner this morning. Messages sent from my iphone would not decrypt on their end but messages from my desktop client did.

 - I'm on Element iOS, 1.6.8 (20211117180457) and Element Web (electron desktop client) 1.9.5
 - They are on Element Android installed from the play store (most recent as of today's date, presumably)
 - We were both on wi-fi the whole time, on all devices involved
 - Restarting either of our apps (without clearing cache) did not fix it
 - @enzingerm's solution (enabling "encrypt to verified sessions only" in the room settings, sending a message, and then disabling it) did fix the problem for new messages sent from my iPhone, but did not immediately cause previously sent messages to decrypt.

Triaging the problem was a bit difficult as it was unclear whether the issue was with my ios app or their android app. I initially went to look in the issues for element ios, as the problem seemed to only be happening with messages from my ios device. This is a high impact bug for user trust of E2E in element and can indirectly affect perceptions of other Matrix apps. I appreciate the persistence of whoever is looking into this 🙂

**Comment by SoftwareSchlosser** [comment]
One thing I noticed is that the _"Waiting for this message, this may take a while"_ message appears on the receiving client as soon as the _"...is typing a message..."_-notification has been received. So that event is already part of the problem and doesn't get decrypted properly.

**Comment by morbificagent** [comment]
Same problem here...

Two android phones. Some days it's working some days this message shows up. Sometimes on one or the other phone...

It's not useable at the moment...

**Comment by morbificagent** [comment]
@enzingerm s Solution does fix this for me too but only for some days and only for the recepients of my messages I think.

And of course I can't switch it on and of always, sent test messages and explain my friends which I have taken with me to Element, that they have to do so 🙄

**Comment by tomasradej** [comment]
I know the devs probably won't appreciate this, but if anyone needs Element to work right now, FluffyChat doesn't seem to have this problem for me, (any messages that can't be decrypted are decrypted after clicking on them) so you can use that client until this issue is fixed.

**Comment by pedymaster** [comment]
It happened to me and my wife few days back again. The interesting thing was, she had the problem (`Waiting for this message, this may take a while`) Only when I wrote her from linux element. If I send the message from windows element or android element, it was OK.
Issue disappeared after about two days. Just pointing out that the source of the message might play some role as well

Wife's client was Element for Android
Mine's client were Element for Linux (arch), Windows 10 and Android as well

sorry I did not catch the versions, but we keep everything updated so something recent

**Comment by BillCarsonFr** [comment]
> It happened to me and my wife few days back again. The interesting thing was, she had the problem (`Waiting for this message, this may take a while`) Only when I wrote her from linux element. If I send the message from windows element or android element, it was OK. Issue disappeared after about two days. Just pointing out that the source of the message might play some role as well
> 
> Wife's client was Element for Android Mine's client were Element for Linux (arch), Windows 10 and Android as well
> 
> sorry I did not catch the versions, but we keep everything updated so something recent

Hello,
What would help us the best is that you send a bug report (rageshake) from your element for linux (the one sending messages that can't be decrypted), as well as from the receiver device; just after the problem occurs.

To send a rageshake, you can go to settings feedbacks or type /rageshake in the chat composer.

Meanwhile from your Element Linux you can type /discardsessoin to restart a new session for future messages

**Comment by morbificagent** [comment]
Here its not about the sending device as the message can be encrypted at the receiver Webinterface and the receiver windows desktop app but not from the android app.

Like this:
(Sender) Sending message from element.io Webinterface or his android app
-> (Receiver) can read it in his element.io Webinterface
-> (Receiver) gets "Waiting for this message, this may take a while" in the android alements app

**Comment by ggogel** [comment]
> Here its not about the sending device as the message can be encrypted at the receiver Webinterface and the receiver windows desktop app but not from the android app.
> 
> Like this: (Sender) Sending message from element.io Webinterface or his android app -> (Receiver) can read it in his element.io Webinterface -> (Receiver) gets "Waiting for this message, this may take a while" in the android alements app
> 
> Have opened a call @element.io for this..

No, we can't be sure about that. The sending client might actually not exchange the keys or the receiving device might not be able to process them in any way. Only because it works on one receiving device it doesn't mean that the bug is on the receiving side.

**Comment by BillCarsonFr** [comment]
If you are curious you can have more details here -> https://www.youtube.com/watch?v=QSeVHiN1dJE

But the sender part is always very important. We have to know if the sender sent the room_key to the receiver, and if not why? Is it because he thinks the receiver is not in the room? or that he can't established the secure olm channel to distribute the key (no more one time key? / blocked?) ?

From the sender logs we can conduct investigations. From the receiver there is not much we can do.

**Comment by morbificagent** [comment]
@BillCarsonFr Many thanks, i will take a look at it when my normal work is done...

Have sent logs to the support with the given NVT# and waiting for response now hoping the best.

**Comment by crtxcr** [comment]
I am part of an encrypted group chat. I am using Element on Linux as well as on Android 12

A person there usually sends from mobile (Android), sometimes several messages in a row. Usually, all are being decrypted immediately, except the last one, for which I have to wait on Android. 
 
However, I don't have to wait with the Linux app. It also doesn't make a difference whether the Element desktop was running or not, it appears to always be decrypting the messages without issues.

No such issues on the sender's end though. 

Force-stopping element and clearing cache does not help in my case.

**Comment by evert** [comment]
Just another confirmation of this issue. We specifically have 2 people in a group chat that are running into this issue often. Just them together, everyone else in the same chats doesn't have the problem

**Comment by equaa** [comment]
Issue happened to me for the 4th time or something along those lines. Tried doing `/discardsession` and logging out on receiving end with no luck ([reacted the same as this user](https://github.com/vector-im/element-android/issues/1721#issuecomment-970231987). Creating a brand new device (i.e. logging in/out from element web) from the sender solved the issue.

**Comment by akshaylive** [comment]
I had this issue happen recently with a two-person group. It started as soon as I linked a new web session and used my phone to verify it. Not sure if it's related.

**Comment by russelldavies** [comment]
I experienced this issue recently between two Android devices and fortunately I was sitting beside the person who was having issues.

I sent a message at 2022-01-13 12:40:23 and noticed it was not decrypting on the other device. I immediately sent two more messages which were undecipherable. I then ran `/discardsession` on my device and sent a message (at 2022-01-13 12:42:36) which could decrypted.

I exported the key request audit logs on both devices, removed extraneous key requests for other rooms, and cleaned it up a bit. Alice is the sender device and Bob is the receiving device. `<session_id_A>` is what the first three messages were encrypted with and `<session_id_D>` is what the fourth, decryptable, message was encrypted with.

Alice:
```
[2022-01-06 18:55:24.344] m.room_key         from:@alice:homeserver - sessionId:<session_id_A> roomId:<room_id> dest:@bob:homeserver|[Bob Android]
[2022-01-06 18:55:24.344] m.room_key         from:@alice:homeserver - sessionId:<session_id_A> roomId:<room_id> dest:@bob:homeserver|[Bob Android]
[2022-01-06 18:55:24.344] m.room_key         from:@alice:homeserver - sessionId:<session_id_A> roomId:<room_id> dest:@bob:homeserver|[Bob Android]
[2022-01-06 18:55:24.344] m.room_key         from:@alice:homeserver - sessionId:<session_id_A> roomId:<room_id> dest:@bob:homeserver|[Bob Android]
[2022-01-06 18:55:27.793] m.room_key_request from:@bob:homeserver   - reqId:<request_id_A> action:request sessionId: <session_id_A> requestedBy: [Bob Android]
[2022-01-06 18:55:31.102] m.room_key_request from:@bob:homeserver   - reqId:<request_id_A> action:request_cancellation requestedBy: [Bob Android]
[2022-01-08 14:40:53.939] m.room_key         from:@alice:homeserver - sessionId:<session_id_B> roomId:<room_id> dest:me
[2022-01-10 19:46:27.495] m.room_key_request from:@alice:homeserver - reqId:<request_id_B> action:request sessionId: NVCnKXVGJzJBtC9T11+By8OPT1frhgY6PbXYNgbUYtc requestedBy: [Alice Desktop]
[2022-01-10 19:46:29.747] m.room_key_request from:@alice:homeserver - reqId:<request_id_B> action:request_cancellation requestedBy: [Alice Desktop]
[2022-01-11 15:53:46.571] m.room_key         from:@alice:homeserver - sessionId:<session_id_C> roomId:<room_id> dest:me
# `/discardsession` run
[2022-01-13 12:42:35.545] m.room_key         from:@alice:homeserver - sessionId:<session_id_D> roomId:<room_id> dest:@bob:homeserver|[Bob Android]
[2022-01-13 12:42:35.545] m.room_key         from:@alice:homeserver - sessionId:<session_id_D> roomId:<room_id> dest:@bob:homeserver|[Bob Android]
[2022-01-13 12:42:35.545] m.room_key         from:@alice:homeserver - sessionId:<session_id_D> roomId:<room_id> dest:@bob:homeserver|[Bob Android]
[2022-01-13 12:42:35.545] m.room_key         from:@alice:homeserver - sessionId:<session_id_D> roomId:<room_id> dest:@bob:homeserver|[Bob Android]
[2022-01-13 12:42:35.545] m.room_key         from:@alice:homeserver - sessionId:<session_id_D> roomId:<room_id> dest:@bob:homeserver|[Bob Android]
```

Bob:
```
[2022-01-06 18:55:27.673] m.room_key_request   from:@bob:homeserver   - reqId:<request_id_A> action:request sessionId: <session_id_A> requestedBy: [Bob Android]
[2022-01-06 18:55:32.030] m.room_key_request   from:@bob:homeserver   - reqId:<request_id_A> action:request_cancellation requestedBy: [Bob Android]
[2022-01-06 18:55:32.522] m.room_key           from:@alice:homeserver - sessionId:<session_id_A> roomId:<room_id> dest:me
[2022-01-06 18:55:32.522] m.room_key_request   from:@bob:homeserver   - reqId:<request_id_A> action:request sessionId: <session_id_A> requestedBy: [Bob Android]
[2022-01-06 18:55:32.522] m.forwarded_room_key from:@alice:homeserver - sessionId:<session_id_A> From Device (sender key):<sender_key>
[2022-01-06 18:55:32.598] m.room_key_request   from:@bob:homeserver   - reqId:<request_id_A> action:request_cancellation requestedBy: [Bob Android]
[2022-01-08 14:43:02.931] m.room_key_request   from:@bob:homeserver   - reqId:<request_id_C> action:request sessionId: <session_id_B> requestedBy: [Bob Android]
[2022-01-08 14:43:03.251] m.room_key_request   from:@bob:homeserver   - reqId:<request_id_C> action:request_cancellation requestedBy: [Bob Android]
[2022-01-08 14:43:03.687] m.room_key           from:@alice:homeserver - sessionId:<session_id_B> roomId:<room_id> dest:me
[2022-01-08 14:47:41.211] m.room_key_request   from:@bob:homeserver   - reqId:<request_id_C> action:request sessionId: <session_id_B> requestedBy: [Bob Android]
[2022-01-08 14:47:41.211] m.forwarded_room_key from:@alice:homeserver - sessionId:<session_id_B> From Device (sender key):qTGr1CpayjSGzA2FXq0aCc2oDieL8hKLgTi3fkH7XDk
[2022-01-08 14:47:41.211] m.room_key_request   from:@bob:homeserver   - reqId:<request_id_C> action:request_cancellation requestedBy: [Bob Android]
[2022-01-08 16:53:58.917] m.room_key_request   from:@bob:homeserver   - reqId:<request_id_D> action:request sessionId: <session_id_B> requestedBy: [Bob Web]
[2022-01-08 16:53:58.917] m.room_key_request   from:@bob:homeserver   - reqId:<request_id_E> action:request sessionId: YXhNHUOwtAXmh/qCl4bsK01kcwuZXHM8Q9J6eHplzyc requestedBy: [Bob Web]
[2022-01-08 16:53:58.917] m.room_key_request   from:@bob:homeserver   - reqId:<request_id_E> action:request_cancellation requestedBy: [Bob Web]
[2022-01-08 16:53:58.917] m.room_key_request   from:@bob:homeserver   - reqId:<request_id_D> action:request_cancellation requestedBy: [Bob Web]
[2022-01-11 15:52:40.349] m.room_key           from:@alice:homeserver - sessionId:<session_id_C> roomId:<room_id> dest:me
# Messages decrypted again
[2022-01-13 12:42:35.197] m.room_key           from:@alice:homeserver - sessionId:<session_id_D> roomId:<room_id> dest:me
```

**Comment by morbificagent** [comment]
Many thanks... i was trying to reproduce this problem too since 2 weeks for generating logs from sender and receiver without luck.

Great you made it.

Hoping that someone can see the problem in it.

**Comment by KeenMaron** [comment]
Same issue here. Partner receives pictures but my text messages produce the problem of the topic.

**Comment by Bug-Ninja** [comment]
Same thing is happening to me on Android. I am completely new to Matrix, so I installed Element from F-droid 1.3.18 [40103180] (F-e5874e43). I created a new account in the matrix.org server.
After email verification I joined 2 rooms. I can see names, avatars, thumbs up and messages of people leaving the room. When I tap one the messages, it says the encryption keys weren't properly sent to me.


![IMG_20220214_163559](https://user-images.githubusercontent.com/17528534/153894905-b87a0848-90be-4ad3-abd5-7c3af5f6e0c3.jpg)

Just to make clear: I have not logged in on any other client before. 
Out if curiousity I logged in in the browser, had to verify my session in the android app en there I saw I couldn't see messages before a specific point in time. Is it an error or is this a setting in the room that new members can't see previous messages? In the last case I guess this should be implemented in a more user friendly way in the Android app.

**Comment by jm-github** [comment]
The problem just occured to me. The message is not displayed in the conversation, but was displayed well in the notification.

**Comment by Yajo** [comment]
I resetted the keys backup in my desktop element. Waited until all keys were backed up. Then, restored those in the phone. It worked.

**Comment by heini** [comment]
@Yajo, were you the one on the sending or on the receiving side?

**Comment by Yajo** [comment]
Receiving

**Comment by egtann** [comment]
This is happening to me in a 2-person group chat, one person on iOS (sending) and another on Android (receiving). The messages do tend to come through but after several hours of "waiting for this message".

**Comment by morbificagent** [comment]
Never ending story.... I am using another client now.
It's a shame that this couldn't be resolved until today.

Arent the element developer using their own client?

**Comment by dead10ck** [comment]
> It's a shame that this couldn't be resolved until today.

@morbificagent "until today"? Was this fixed?

**Comment by morbificagent** [comment]
Oh I'm sorry. It was only my bad English.
It isn't fixed I think.

**Comment by wytchmaster** [comment]
https://www.encrypted.at/element-matrix-chat-fix-waiting-for-this-message-this-may-take-a-while/ works for me

**Comment by dodedodo** [comment]
For those wondering, wytchmaster is talking about the `/discardsession` command. This has already been discussed a couple of times. It does not seem to work for everyone. The article itself actually points back to [this comment](https://github.com/vector-im/element-android/issues/1721#issuecomment-880214682)

**Comment by eTomte** [comment]
This just happened to me again today, after not happening for six months.
the  /discardsession workaround worked again, but I'm on latest synapse and latest Play store android client, and still happening.

**Comment by piiot** [comment]
Similarly to eTomte, this issue again happened to me today after being dormant for 6+ months.  Only seeing this on the Android client, and the client along with synapse are on the latest version.

**Comment by NHAS** [comment]
None of these solutions worked. I have to logout and log back in to my account each time. And it reoccurs frequently making matrix borderline unusable. 
I use this every day, on the latest version of element and synapse.

**Comment by austinbutler** [comment]
Happened to me again today. Android to Android (both Element).

Periodically I consider moving friends from Messenger and Google Chat to Element, but core critical issues like this remind me it's not ready yet. I can deal with lack of polish and little fun UI features that Element lacks compared to the competition, but messages not being delivered is a complete deal breaker. Yet this issue appears to be tagged with "papercuts." And it appears to be being treated as a papercut. It's not. It's a core function and it's critical.

**Comment by austinbutler** [comment]
If there's someone at Element that needs someone technical to reach out to for logs I'll volunteer my time.

**Comment by jm-github** [comment]
Happening again to another contact in an encrypted room, on Android phone.
The message shows well in the notification, but not in the room itself. The message only shows well when the same contact sends another message, which is then not readable.

I also volunteer to help fixing this bug.
Maybe we could first create an encrypted room to try reproducing it?

**Comment by morbificagent** [comment]
I have a call open now for month. And the only thing which is happening is that i was asked for logs one time.
Now i receive Mails every month with the question if the problem is resolved... NO IT ISNT.
But it looks like nobody knows whats the Problem or what to do now.
Arent they using their own product?

Instead of throwing all energy in getting element stable and useable i receive app-updates with "a new audio-player"... WTF?

So @austinbutler you are right. 

Element isnt ready.

**Comment by BillCarsonFr** [comment]
Hello, here is a link to the global issue tracking decryption problems https://github.com/vector-im/element-meta/issues/245

You can find more details about this issue there.
The important part is that in order to debug an issue we need the logs from both the sender and recipient (see linked issue to see how to submit debug logs).

Several root causes have been fixed, and more are going to be fixed with time. Complete bug reports with both debug logs are helping a lot, thx again.

**Comment by BillCarsonFr** [comment]
> 

Hello @jm-github , did you sent a rageshake?

**Comment by austinbutler** [comment]
Thanks @BillCarsonFr. Just submitted bug reports from both sender and receiver.

**Comment by BillCarsonFr** [comment]
> Thanks @BillCarsonFr. Just submitted bug reports from both sender and receiver.

THx got them.

**Comment by jm-github** [comment]
> > 
> 
> Hello @jm-github , did you sent a rageshake?
The bug is not happening on my side. I will try to ask the 2 persons involved in the bug to send one

**Comment by jm-github** [comment]
> > 
> 
> Hello @jm-github , did you sent a rageshake?

You should have received another pair of rageshakes. I hope it will help

**Comment by austinbutler** [comment]
This has cropped up every I dunno 6 months or so for a few messages at a time so it was more tolerable, but since I posted 3 days ago the issue is happening to me very frequently. This is the worst it has been, Element is now not usable for me. Weirdly it works perfectly fine when I view the exact same message on desktop (just after it was sent). I also don't get a notification for these "waiting" messages, so I don't even get notified someone tried to contact me. I wonder how the explanation of "encryption keys were not properly sent to you" makes sense given the same sender's messages show up just fine on desktop?

Anyone else having this issue show up more frequently since the last week?

**Comment by lindhe** [comment]
Nope. I haven't had it since November. Strange that it seems so random...

**Comment by KeenMaron** [comment]
The problem occurs in my synapse network randomly too. From a dev viewpoint it's quite hard bug. There must be a way to systematically provoke the bug. Random occurring bugs are a nightmare...

**Comment by morbificagent** [comment]
Could you @BillCarsonFr and/or the Crypto-team find something in the last logs sent? Would be great to get informed here about the results to get the feeling someone is working on this. 

I have bought the service/vserver or whatever from element.io and thought i would get some sort of "premium messenger" and now i have to use it with the matrix-client "FluffyChat" (man i hate this name ;-) ) because of the official element client isnt working on mobile devices.

**Comment by austinbutler** [comment]
Yeah still happening for me. Literally have five "Waiting for this message" in Element Android while simultaneously looking at the encrypted version of those same messages on Element Desktop and FluffyChat Android. So I can confirm that FluffyChat is a good workaround. If it's a server issue (note I am using matrix.org), then it's a server issue that everything else handles just fine and Element Android does not.

**Comment by rajil** [comment]
I had this happen to a user who was connecting to my server. I am just wondering whether this can happen when an upgrade of synapse is happening on the server? Maybe the client gets into a messy state because of that.

**Comment by heini** [comment]
I also had this a couple of days ago, but during verification (between two users with element on Android, both had 1.4.25 installed). One user started the verification process, but the other one didn't receive a notification to accept it. It only worked when the first user started the verification from FluffyChat.

**Comment by kpfleming** [comment]
I have had this happening in a 1-to-1 chat for a couple of weeks now. The other user and I are the only users on our Synapse server (although it is federated, but we are the only local users). Same as above, when the other user sends a new message, the previously unavailable message appears, and the new one does not.

Both of us are using Android Element from the Play Store, and as noted above if we use Element Desktop to look at the same chat all of the messages appear normally.

**Comment by benneti** [comment]
> Both of us are using Android Element from the Play Store, and as noted above if we use Element Desktop to look at the same chat all of the messages appear normally.

@kpfleming I had a very similar problem, what fixed it for me was to delete the session keys on the element desktop clients (sessions of the person with the problems) and it did not happen again since several month.

**Comment by jm-github** [comment]
> 

Does the notification contains the message in clear text and then it is not visible in the conversation ?

**Comment by heini** [comment]
Better don't do that. Afterwards you need to re-verify all your session AND contacts...

**Comment by benneti** [comment]
> Better don't do that. Afterwards you need to re-verify all your sessions AND contacts...

Ah yes sorry for not being clear, what I meant was "/discardsession" in the chat with problems, which if I remember correctly did not make re-verification necessary

**Comment by heini** [comment]
Ah, OK. Thanks for clarification.

**Comment by kpfleming** [comment]
@jm-github For some reason I'm not getting notifications for this chat, even though they are enabled in every screen that has notification controls, so I can't answer about whether the notification contains the message text.

**Comment by kpfleming** [comment]
In order to try to fix the notifications issue I uninstalled Element from my phone and reinstalled it. Logged in, verified the session using my laptop session. There are four 'direct' chats open.

Two of them backfilled all content *except* the last message from the other party; that last message says "waiting". The other two backfilled my messages, but all messages from the other party say "waiting". One of those other parties is another user on my homeserver, the other is a user on a federated server.

**Comment by rajil** [comment]
As a HS owner of a family group, is it possible to find out which client or user may be stuck because of this. It will allow me to reach out of band and ask them to reinstall the client.

**Comment by jamtwister** [comment]
I had the same issue today. I added a new device. Crosssigning is enabled and e2e keys backup too.
After verifying the new session some messages are shown, some others not (waiting ...).
Then I restored e2e keys from backup manually, which was successful. Then all messages are shown correctly decrypted.
Hopefully this can help someone.

**Comment by KeenMaron** [comment]
Still, it shouldn't be that complicated and it is an issue which should be fixed. You can't maintain 20+ users because of this problem 😕

**Comment by ahihi** [comment]
i also started seeing this yesterday and submitted a rageshake about it. i can confirm that restoring keys from backup seems to fix it! thanks @jamtwister

**Comment by rmgr** [comment]
> > 
> 
> Does the notification contains the message in clear text and then it is not visible in the conversation ?

I did notice this happen to me. The notification contained the message but when I opened it I can't see the message in the chat.

**Comment by rajil** [comment]
I had this happen today in a 1:1 chat. Both clients are using 1.4.27 version.

**Comment by freli83** [comment]
I had the same problems it appears more and more often in 1:1 chat, but also sometimes on group chats.
Both mobile devices have the current version 1.5.1, Server Version is: matrix-synapse-py3/unknown,now 1.68.0+buster1
I have this problem only on the mobile App not on the desktop client!

The mobile device where the problem occurs at the most, has only one verified session no other devices are used. The session is verified an valid.

Sometime it helps to re import the End-to-End keys.
I have found that if the app is closed completely, then restarts the server and then reopens the app the messages arrive normally.

I hope it helps to find and fix this error. 
It's too often no communication possible, which makes the app unfortunately useless.

**Comment by vtronko** [comment]
This issue really becomes annoying, it started happening all the time on latest (1.5.1) version in 1:1 chats. It's impossible to decipher the content of the message even after new messages randomly start to arrive normally. This is really unacceptable for crucial communications and renders application useless without backup communication channel.

**Comment by freli83** [comment]
I noticed that you can read the message via the "pop-up" function and then not within the app. Than you get the message `Waiting for this message, this may take a while.`. So maybe it's not a problem with the keys?

**Comment by KeenMaron** [comment]
But why is it still happening? Why is this workaround with clearing cache still necessary ? Is it not possible to fix it completly?

**Comment by freli83** [comment]
> In case the message is readable in the notification you can kill the app and it should be visible when you relaunch. And if it persists, does clear cache&reload help?

Sometimes, in the most cases it helps to see some messages but not all. Often the most messages are then visible except the last 1 or 2. Restart the Server and relaunch the App on the phone helpd more in my case.

**Comment by SoftwareSchlosser** [comment]
Just wanted to mention that I didn't see this for several months and it started happening again a few days ago. Might be a coincidence but to me it seems like it was fixed (maybe unintentionally?) and now it's broken again.

**Comment by pedymaster** [comment]
I have the same experience as @SoftwareSchlosser

**Comment by brandsimon** [comment]
Since 1.5.2 I enabled `MSC3061: Sharing room keys for past messages`. The app immediately restarted and reloaded all messages. Now a few messages, which were perfectly readable before, show the error `Waiting for this message, this may take a while` or `Encrypted message`. When long-clicking some messages, I am able to copy the text and then I can paste the complete plaintext (also for very long messages). It seams that this behavior only happens on edited messages, so I am not sure if this is related to the problem described here or a new one.

**Comment by KeenMaron** [comment]
Update: except one user nobody can still read the message of the user with this issue. i excluded him from the group and tried to re-invite but he gets no notification and can't join. What a mess😂

**Comment by amontalenti** [comment]
I hit this issue recently when upgrading my Android phone. On my desktop client(s) on macOS and Linux, I'm able to see my encrypted DMs no problem. But on the new Android phone, all my encrypted DMs had all their messages blocked, going back in history, with the "Waiting for this message" note. I then investigated a little and found Element's listing of my Sessions, and noticed that one of the Android sessions was "unverified". So, I signed out all the unused sessions associated with my account, including that unverified one. Intriguingly, that led the DMs on my Android client to start populating, _but not all of them_. I would say about 50% of the DMs decrypted, but the other half still showed the "Waiting for this message", notice. What's more, all of the historical messages that managed to successfully decrypt now had a gray shield icon next to them, which said, upon long-press, "The authenticity of this encrypted message can't be guaranteed on this device." Yet, from my desktop clients, those same messages show up without the shield icon.

I hope this information helps the devs get to the bottom of what's going on. I would also appreciate any ideas about how to get those remaining messages decrypted on my Android client!

**Comment by PC-Admin** [comment]
Observed this bug recently, only seems to happen in Element Android. It's very strange since 95%+ messages transmit and load properly, but then every once and a while you get a message that doesn't load and you see this "Waiting for this message, this may take a while." message.

**Comment by amontalenti** [comment]
Interestingly, I waited a couple days after removing the unverified sessions from my server/account. Eventually, those messages with the "Waiting for.." notice populated in. But all my historical messages in these DMs are now marked with the gray shield icon. So, I think there is still some sort of bug here, but at least the messages did decrypt on Android eventually. Hmm.

**Comment by ccarter-tmt** [comment]
This is happening to me to, on the latest Android release of Element. Windows PC desktop Element does not exhibit the issue. Would like to know a fix/workaround as we have two mobile users here who cannot communicate due to this problem...

**Comment by clicit** [comment]
Same here since the last update. We cannot really reproduce the problem, it's happening sometimes with some users and different results - meaning that some message finally decrypt after a couple of minutes/hours/days, but some don't. Please really put high prio on this! Thanks

**Comment by gabrc52** [comment]
![Screenshot_20230101-153451](https://user-images.githubusercontent.com/24363938/210185278-bc79aaae-24cf-4ac8-91d2-f48e36cdf1ea.png)

It just happened to me when my session timed out and I had to log in again

**Comment by gabrc52** [comment]
Hm, I cleared my account and reverified and messages are coming up. I'll try to get logs next time it happens, if it does.

**Comment by ccarter-tmt** [comment]
> Hm, I cleared my account and reverified and messages are coming up. I'll try to get logs next time it happens, if it does.

What does 'cleared my account' mean?

**Comment by karlpip** [comment]
Happening for me too.

**Comment by stuart12** [comment]
I fixed the same problem by upgrading Element (to f-droid 1.5.26). The upgrade was not a major upgrade so I do think that new version fixed any bugs. I imagine that it was just the action of doing an upgrade started the process of downloading old keys.

**Comment by Patriot2407** [comment]
I always upgrade all my F-droid as soon as there is an update, I still have issues. I haven't had the issue since I last posted, but it has been intermittent. The fix for me is always to completely clear out my login info in the f-droid app and relog in

**Comment by Buntbart** [comment]
I had this issue a few days ago again, with the latest version from F-Droid installed on both sides and both accounts on matrix.org.
It is usually solved by typing `/discardsession` on the sending side.

**Comment by Patriot2407** [comment]
This does work so that new messages go through, however the messages that won't send never do, and that makes this chat unreliable, which is unacceptable. I have no way of knowing if messages all of a sudden stop sending unless a user manually tells me, and it makes for a terrible experience. To the sender it appears that the receiver has actually received the content and read it, no error ever appears.

**Comment by KeenMaron** [comment]
i just wondering why this heavy bugs are so long persistent in the software...

**Comment by austinbutler** [comment]
> i just wondering why this heavy bugs are so long persistent in the software... It's almost 3 years.

It's labeled as a papercut for one thing. Despite this making Element completely unusable for encrypted chat.

**Comment by KeenMaron** [comment]
sorry i don't get it, what do you mean with "labeled as papercut" ?

**Comment by austinbutler** [comment]
![image](https://user-images.githubusercontent.com/354741/227332083-33d8ea57-81d7-4c5c-b382-c71ad02303e2.png)

**Comment by austinbutler** [comment]
For the record this is the most-commented and third-most reacted issue for this repo.

**Comment by florian6973** [comment]
Still having this bug after several months, I am sad to say but this is the reason why I stopped using Element. I have always supported the initiative and project but clearly this app is not reliable and developers are not on the right subjects to make it work. When people ask what do I think about Element, I am forced to tell them that even sending messages is just a pain.

**Comment by Ulrar** [comment]
Not to pile on but that indeed has become a big problem lately, it happens often and I've had messages in that state for weeks, and the only solution we found is to just resend the message through another mean (WhatsApp for example), which is terrible

**Comment by johncarterofmars** [comment]
I may be able to add some more to this. We have been testing with about 20 users, 40+ devices. These are a mix of Windows, Linux desktop apps, Android, and iOS. This happens from Android users and iOS users. It is seemingly when those users have weak internet access. If they are on Wifi in the office, it never happens. If they are using data and maybe it's not the best connection, it happens. The messages are never decrypted, even after a week. 

On the desktop app, it says: Unable to decrypt message. 
It asks to resend the keys from another session but even after doing this, nothing changes. 

From our experience, it seems to be when the sending user has a poor internet connection. 

Not sure if this helps or not.

**Comment by thebiblelover7** [comment]
Agreed. I have seen the same thing. When I'm out and about, and the connection isn't the best, the message will send very slowly and the other person will fail to see my message (with the error of "Waiting for this message, this may take a while"). Notice though, that I can see the message totally fine from my other devices.

**Comment by clicit** [comment]
Same here. International team with significantly higher error rate when traveling abroad with a poor Internet connection.

**Comment by Patriot2407** [comment]
Yes, that is a common sympton for me as well... Now we need a fix. Other chat programs don't stop working just because we are on a cell tower.

**Comment by johncarterofmars** [comment]
I have a little more to add today. Yesterday, one user sending from an Android device developed this issue. The strange part was that while Linux and Windows workstations could read the messages, other Android devices could not. Even now, after 24 hours, and the Android client still says "Waiting for this message". If I double-click on that message, it tells me that the keys weren't properly sent to me. I don't understand how they were properly sent to the desktop clients and not the Android clients.

**Comment by clicit** [comment]
Massive problems again and again as soon as people travel to other countries...

One more detail we observed several times: 
User A is out of country and receiving messages in a group from User B and also directly from User B. The messages in the group cannot be decrypted, while the direct messages are decrypted successfully.

Also interesting the online/offline reproduction approach described in this related issue:  #8245 
.. since having lots of disconnects while travelling this sounds reasonable

**Comment by TBreiteneder** [comment]
Having issues with this all the time.
Group chats have a user where I cannot read the messages form a certain timeframe, stating the keys could not be decrypted.
It tells me to chek on another device, where I get Waiting for this message, this may take a while

**Comment by KeenMaron** [comment]
Sorry but the whole encryption functionality is kind of a "mess". There are so many different problems that makes element with matrix unuseable for a non technical person who just want be a user. The idea it works out of the box is really far far away. I hope with Element R or X it will be way better, but in my expierience big updates of software always bring new bugs which could make everything worse.

**Comment by morbificagent** [comment]
Use the matrix client "Fluffy" instead if you are using android.
The name and the logo are... ok, not professional... but the client itself is working much better as the element app and without this all-time favorite bug.

3 Years after this massive bug came up there is still no solution. So finish with this. it wont get fixed.

**Comment by Patriot2407** [comment]
> Use the matrix client "Fluffy" instead if you are using android.
> The name and the logo are... ok, not professional... but the client itself is working much better as the element app and without this all-time favorite bug.
> 
> 3 Years after this massive bug came up there is still no solution. So finish with this. it wont get fixed.

Project says it's frozen, also no way to change to my local server that I can see, and yes it looks very unprofessional and silly.

**Comment by thomaspeissl** [comment]
> Project says it's frozen
AFAIK they moved to GitLab https://gitlab.com/famedly/fluffychat

**Comment by red0888** [comment]
Have the same issue, this is unacceptable and terrible!

**Comment by JokerGermany** [comment]
It's important to tell if you use Version 1.6.2 and therefore the new rust crypto

**Comment by KeenMaron** [comment]
Since when is this update out ?

**Comment by JokerGermany** [comment]
Last week
https://github.com/vector-im/element-android/releases/tag/v1.6.2
, but i don't know if it's already in the playstore

**Comment by austinbutler** [comment]
Wow finally! Though the Play Store version is three releases behind. F-Droid, too. I wonder why?

Anyone not wanting to wait should be able to install directly from the releases link above. But the fact that they don't seem to keep the app up to date makes.me wonder if they don't consider these releases stable yet?

**Comment by austinbutler** [comment]
Oh they touch on this in the README: https://github.com/vector-im/element-android/blob/develop/README.md#releases-to-app-stores

**Comment by KeenMaron** [comment]
So how long do we have to wait until it is published in the Playstore?

**Comment by benmordecai** [comment]
@KeenMaron that is addressed in the README above. It's an indefinite amount of time because Google is unpredictable with how long it takes to approve updates.

**Comment by JokerGermany** [comment]
Perhaps its another story:
https://matrix.to/#/!AZozoWghOYSIAfaZjJ:matrix.org/$1AwZxtcePbIXxxyfMx5NKNCwcqrSm4VZ441-jCKQz7U?via=matrix.org&via=tchncs.de&via=envs.net
>  don't know about the beta channel, but stable is at 1.5.32 on the Play Store. AFAIK there are still some performance issues with 1.6.2, so I presume that's what is holding up the release (I'm just a lurker though, I'm not involved with development in any way).

**Comment by RussellAult** [comment]
> It's important to tell if you use Version 1.6.2 and therefore the new rust crypto

I just ran into this issue again (for the first time in a while) on 1.6.3 from the Play Store, so the new Rust crypto hasn't totally resolved it. I'm not sure if it's related, but the client in question was also suffering from #6490 , which I resolved by enabling cross-signing (even though the user only ever uses the one client).

**Comment by aelmahmoudy** [comment]
Why isn't it on F-Droid ?

**Comment by benmordecai** [comment]
@RussellAult - is your device producing the messages that fail to decrypt on other users devices? My experience is that 1.6.2 fixed the problem of users creating messages that fail to decrypt but users who have not updated still may produce messages that fail to decrypt when read from a 1.6.2+ device

**Comment by RussellAult** [comment]
> @RussellAult - is your device producing the messages that fail to decrypt on other users devices? My experience is that 1.6.2 fixed the problem of users creating messages that fail to decrypt but users who have not updated still may produce messages that fail to decrypt when read from a 1.6.2+ device

Ah, interesting. The message that failed to decrypt was sent by me from the nheko client v0.10.2 (i.e. the latest version available in Debian's oldstable-backports); this is a bit behind the current release version (0.11.3), but I'm not seeing anything in the release notes that would indicate a major change in the way it handles encryption. Having said that, my messages from nheko are showing up on my phone as "The authenticity of this encrypted message can't be guaranteed on this device" despite the session showing up as "Trusted", so clearly something is up.

**Comment by KeenMaron** [comment]
I'm hosting a Matrix server, and this issue appeared quite often when some users had a bad internet connection. But luckily, we haven't had any problems for a few months. 
Now we have all updated to the latest Android version (1.6.3) hoping this issue will never appear again. One user is using an iPhone, and the Element version is 1.10.14. Out of nowhere, the issue appeared again. His messages are unreadable in one room (can't be encrypted). The funny thing is, direct chat works, and chatting with him in other rooms works as well. How can we fix it?

**Comment by KeenMaron** [comment]
![image](https://github.com/vector-im/element-android/assets/52194699/38a742db-55dd-4937-8525-0b6b1f398e23)
Thats the meta info of the message which cannot be decrypted. Any chance to get this fixed or i must delete the whole room and create a new one?

**Comment by CKwan-hub** [comment]
Encountering the same issue on newest version

**Comment by rajil** [comment]
I was hit by this in 1.6.3 version, sign out/in, import of recovery key did not fix the problem. New messages are now coming in but old messages still have 'waiting message'.

**Comment by Tecol87** [comment]
Are there any production ready alternatives?

**Comment by zoj613** [comment]
I've been having this issue where this message has been showing for months on one of my devices. No matter how long I wait it is still showing after months and therefore cannot read earlier messages prior to installing the app on the devices even after successful verification. I'm using version 1.5.32

**Comment by zoj613** [comment]
> You can try to restore the keys from settings > security and privacy > restore keys. That helped for someone who was stuck after an update to 1.6.3 and the new encryption thingy. While all others had a second session with which they could very the new session created by the rust encryption and 1.6.3, the users with only a single session are stuck.

That worked like a charm. Thank you!

**Comment by saschanaz** [comment]
For me the restore keys way didn't work, but somehow logging in and verifying on another device magically fixed the issue.

**Comment by Tecol87** [comment]
Since 1.6.5, I haven't had any issues anymore.

**Comment by ismay** [comment]
I encountered this whilst being logged in to element android, element web and android fluffychat (all the same account). I had already uninstalled fluffychat. End the fluffychat session via element web's session management UI resolved this bug for me.

Might be worth checking if you have stale sessions and killing them if you encounter this bug. Seemed to fix it for me.

**Comment by richvdh** [comment]
Duplicate of https://github.com/element-hq/element-meta/issues/245
